making no-archive the default and adding the --archive / -a flag

rid-dim opened the pull request:
### Description

Since I'm doing single file uploads repeatedly atm (uploading friends themes, background images, JS Webcomponents I want to include in other WebApps, ...) where I always just use the datamap and the chunks of the file but never the archive I created this mod for me and from what I hear there's multiple people out there who would like this feature too.

Archives have their place and are good for folder upload/download/syncing ... but if I just want an object or content online on autonomi there is absolutely no reason to create and upload an archive too.

---

side note: I tested downloading a file from raw datamap and the client already supports it - so there doesn't seem to be a breaking change on the other side that would prevent changing default behavior imho

### Related Issue

only general complaints / usability issues around not being able to upload content without creating the archive with the ant client

### Type of Change

Please mark the types of changes made in this pull request.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Other (please describe):

### Checklist

Please ensure all of the following tasks have been completed:

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I've tested locally with my changes.
- [x] I have updated the documentation accordingly.
- [x] I have tried to add commit messages with [commitlint](https://commitlint.js.org/#/).

<!--
### Footer (Hidden from GitHub view)
This template uses [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). Please ensure your commit messages follow these guidelines for clarity and consistency.

Commit messages should be verified using [commitlint](https://commitlint.js.org/#/).

Commit Message Format:
<type>[optional scope]: <description>
[optional body]
[optional footer(s)]

Common Types:
- feat
- fix
- docs
- style
- refactor
- perf
- test
- build
- ci
- chore
- revert
-->

jacderida commented:
Awesome, thanks for the contribution! Would you mind squashing your commits into one please?

jacderida commented on ant-cli/src/commands.rs:
```diff
@@ -70,6 +70,9 @@ pub enum FileCmd {
         /// Upload the file as public. Everyone can see public data on the Network.
         #[arg(short, long)]
         public: bool,
+        /// Create and upload archive after file upload. This enables traditional archive functionality.
```
Could we perhaps state the purpose of the archive here? I think it's for storing metadata.

We could also mention here that it will cost more for the archive.

rid-dim replied:
I've added a more lengthy description - not sure it's too long now - but might not be an issue since I think it clearly states now what happens (which is good imho)

good thought - nobody new to this will have an idea what the archive flag is about

rid-dim committed:
feat(ant-cli): add --archive flag to skip archive creation

- Add --archive option to file upload command (and making non-creation of archives the default behavior)
- creates archives only when flag is set and not without being requested anymore
- Update documentation in README.md

grumbach commented:
Partially covered in #3017
- kept archives as default but add `--no-archive` flag to avoid them
- single files are not archives by default!
